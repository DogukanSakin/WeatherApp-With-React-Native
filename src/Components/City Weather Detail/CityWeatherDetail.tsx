import React,{FC} from 'react';
import {  Text,View,Image} from 'react-native';
import {styles} from './CityWeatherDetail.style';
import Icon from 'react-native-vector-icons/MaterialCommunityIcons';
interface ICardProps{
    darkMode: boolean;
    city: any;
}
const CityWeatherDetailCard:FC<ICardProps>=({darkMode,city})=>{
    const cardColors = styles(darkMode);
    return(
        <>
        <View style={cardColors.weatherDetailContainer}>
            <Icon name='earth' size={20} color='#5BA7FB' testID='weather-detail-card-earth-icon'></Icon>
            <Text style={cardColors.weatherDetailText} testID='weather-detail-card-country-text'>{city.country}</Text>
        </View>
        <View style={cardColors.weatherDetailContainer}>
            <View style={cardColors.weatherDetailCardContainer}>
                <Icon name='eye' size={20} color='#5BA7FB'></Icon>
                <Text style={cardColors.weatherDetailText} testID='weather-detail-card-visibility-text'>Visibility {city.visibility} KM</Text>
            </View>
            <View style={cardColors.weatherDetailCardContainer}>
                <Icon name='thermometer' size={20} color='#5BA7FB'></Icon>
                <Text style={cardColors.weatherDetailText} testID='weather-detail-card-feels-text'>Feels {city.feelsLike_c}Â°C</Text>
            </View>
        </View><View style={cardColors.weatherDetailContainer}>
                <View style={cardColors.weatherDetailCardContainer}>
                    <Icon name='waves' size={20} color='#5BA7FB'></Icon>
                    <Text style={cardColors.weatherDetailText} testID='weather-detail-card-humidity-text'>Humidity {city.humidity} KM</Text>
                </View>
                <View style={cardColors.weatherDetailCardContainer}>
                    <Icon name='weather-windy' size={20} color='#5BA7FB'></Icon>
                    <Text style={cardColors.weatherDetailText} testID='weather-detail-card-wind-text'>Wind {city.wind} KM</Text>
                </View>
        </View></>
    )
}
export default CityWeatherDetailCard;
