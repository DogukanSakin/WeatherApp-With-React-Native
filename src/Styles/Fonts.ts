const Fonts={
    lightFont:'OpenSans-Light',
    regularFont:'OpenSans-Regular',
    semiBoldFont:'OpenSans-SemiBold',
}
export default Fonts;