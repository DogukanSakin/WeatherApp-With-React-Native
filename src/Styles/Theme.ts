export const darkTheme={
    defaultColor:'#161616',
    greyColor:'#313030',
    darkerGreyColor:'#E3E3E3',
    componentColor:'white'
}
export const lightTheme={
    defaultColor:'white',
    greyColor:'#F5F4F4',
    darkerGreyColor:'#424040',
    componentColor:'#161616',
}