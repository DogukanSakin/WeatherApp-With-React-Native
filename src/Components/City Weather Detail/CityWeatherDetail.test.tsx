import React from 'react';
import {render} from '@testing-library/react-native';
import CityWeatherDetail from './CityWeatherDetail';
import { darkTheme,lightTheme } from '../../Styles/Theme';
const testCity={
    country:'TestCountry',
    visibility:10,
    feelsLike_c:15.5,
    humidity:5,
    wind:20
}
test('should match with snapshot with testCity', () => { 
    const comp=render(<CityWeatherDetail darkMode={false} city={testCity}/>);
    expect(comp).toMatchSnapshot();
})
test('should  render cityInfo correctly', () => { 
    const comp=render(<CityWeatherDetail darkMode={false} city={testCity}/>);
    //Country Text:
    const countryText= comp.getByTestId('weather-detail-card-country-text').children[0];
    expect(countryText).toBe(testCity.country);
    //Visibility:
    const visibilityText= comp.getByTestId('weather-detail-card-visibility-text').children.join('');
    expect(visibilityText).toContain(testCity.visibility);
    //Feels:
    const feelsText= comp.getByTestId('weather-detail-card-feels-text').children.join('');
    expect(feelsText).toContain(testCity.feelsLike_c);
    //Wind
    const windText= comp.getByTestId('weather-detail-card-wind-text').children.join('');
    expect(windText).toContain(testCity.wind);
    //Humidity
    const humidityText= comp.getByTestId('weather-detail-card-humidity-text').children.join('');
    expect(humidityText).toContain(testCity.humidity);
})
test('should text color be white if theme is light ', () => { 
    const comp=render(<CityWeatherDetail darkMode={false} city={testCity}/>);
    //Country
    const countryText= comp.getByTestId('weather-detail-card-country-text').props;
    expect(countryText.style['color']).toBe(lightTheme.componentColor);
    //Visibility:
    const visibilityText= comp.getByTestId('weather-detail-card-visibility-text').props;
    expect(visibilityText.style['color']).toBe(lightTheme.componentColor);
    //Feels:
    const feelsText= comp.getByTestId('weather-detail-card-feels-text').props;
    expect(feelsText.style['color']).toBe(lightTheme.componentColor);
    //Wind
    const windText= comp.getByTestId('weather-detail-card-wind-text').props;
    expect(windText.style['color']).toBe(lightTheme.componentColor);
    //Humidity
    const humidityText= comp.getByTestId('weather-detail-card-humidity-text').props;
    expect(humidityText.style['color']).toBe(lightTheme.componentColor);
    
})
test('should text color be white if theme is dark ', () => { 
    const comp=render(<CityWeatherDetail darkMode={true} city={testCity}/>);
    //Country
    const countryText= comp.getByTestId('weather-detail-card-country-text').props;
    expect(countryText.style['color']).toBe(darkTheme.componentColor);
    //Visibility:
    const visibilityText= comp.getByTestId('weather-detail-card-visibility-text').props;
    expect(visibilityText.style['color']).toBe(darkTheme.componentColor);
    //Feels:
    const feelsText= comp.getByTestId('weather-detail-card-feels-text').props;
    expect(feelsText.style['color']).toBe(darkTheme.componentColor);
    //Wind
    const windText= comp.getByTestId('weather-detail-card-wind-text').props;
    expect(windText.style['color']).toBe(darkTheme.componentColor);
    //Humidity
    const humidityText= comp.getByTestId('weather-detail-card-humidity-text').props;
    expect(humidityText.style['color']).toBe(darkTheme.componentColor);
    
})
