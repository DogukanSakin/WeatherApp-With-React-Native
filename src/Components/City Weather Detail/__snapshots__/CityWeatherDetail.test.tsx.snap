// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match with snapshot with testCity 1`] = `
Array [
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
        "marginTop": 15,
      }
    }
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        Array [
          Object {
            "color": "#5BA7FB",
            "fontSize": 20,
          },
          undefined,
          Object {
            "fontFamily": "Material Design Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
      testID="weather-detail-card-earth-icon"
    >
      󰇧
    </Text>
    <Text
      style={
        Object {
          "color": "#161616",
          "fontFamily": "OpenSans-Regular",
          "fontSize": 15,
          "marginRight": 15,
          "paddingLeft": 5,
        }
      }
      testID="weather-detail-card-country-text"
    >
      TestCountry
    </Text>
  </View>,
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
        "marginTop": 15,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
          "marginBottom": 10,
        }
      }
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          Array [
            Object {
              "color": "#5BA7FB",
              "fontSize": 20,
            },
            undefined,
            Object {
              "fontFamily": "Material Design Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        󰈈
      </Text>
      <Text
        style={
          Object {
            "color": "#161616",
            "fontFamily": "OpenSans-Regular",
            "fontSize": 15,
            "marginRight": 15,
            "paddingLeft": 5,
          }
        }
        testID="weather-detail-card-visibility-text"
      >
        Visibility 
        10
         KM
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
          "marginBottom": 10,
        }
      }
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          Array [
            Object {
              "color": "#5BA7FB",
              "fontSize": 20,
            },
            undefined,
            Object {
              "fontFamily": "Material Design Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        󰔏
      </Text>
      <Text
        style={
          Object {
            "color": "#161616",
            "fontFamily": "OpenSans-Regular",
            "fontSize": 15,
            "marginRight": 15,
            "paddingLeft": 5,
          }
        }
        testID="weather-detail-card-feels-text"
      >
        Feels 
        15.5
        °C
      </Text>
    </View>
  </View>,
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
        "marginTop": 15,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
          "marginBottom": 10,
        }
      }
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          Array [
            Object {
              "color": "#5BA7FB",
              "fontSize": 20,
            },
            undefined,
            Object {
              "fontFamily": "Material Design Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        󰞍
      </Text>
      <Text
        style={
          Object {
            "color": "#161616",
            "fontFamily": "OpenSans-Regular",
            "fontSize": 15,
            "marginRight": 15,
            "paddingLeft": 5,
          }
        }
        testID="weather-detail-card-humidity-text"
      >
        Humidity 
        5
         KM
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
          "marginBottom": 10,
        }
      }
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          Array [
            Object {
              "color": "#5BA7FB",
              "fontSize": 20,
            },
            undefined,
            Object {
              "fontFamily": "Material Design Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        󰖝
      </Text>
      <Text
        style={
          Object {
            "color": "#161616",
            "fontFamily": "OpenSans-Regular",
            "fontSize": 15,
            "marginRight": 15,
            "paddingLeft": 5,
          }
        }
        testID="weather-detail-card-wind-text"
      >
        Wind 
        20
         KM
      </Text>
    </View>
  </View>,
]
`;
